### PR TITLE
Add Answer Button and functionality

### PR DESCRIPTION
**Pull request by iwantmyhatback**
- Created "AddAnswer" component for inside of modal
- Put in place axios request to post answers to the API
- Passed product_id into "AddAnswer" because it was required to reload questions
- Passed "getProductQuestions" (dispatch function) to reload Q&A after posting
- added form validation and restrictions on input length in the "addAnswer" module

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ function App(props) {
         <ReviewsListContainer />
       </div>
       <div className="container">
-        <QuestionListContainer />
+        <QuestionListContainer paramsId={params.id} />
```
Comment (dprevoznik) [inline src/App.js:25]: Great. I am glad this was so easy to implement in react router.

File: src/components/questions/addAnswer.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      validEmail: false
+    }
+    this.onChange = this.onChange.bind(this);
+    this.submitAnswer = this.submitAnswer.bind(this);
+    this.checkEmailValidity = this.checkEmailValidity.bind(this);
+  }
+
+  onChange(event){
+    console.log(event.target.value);
+    let feild = event.target.name;
```
Comment (dprevoznik) [inline src/components/questions/addAnswer.js:24]: we spoke about this spelling error. It will be fixed on next commit.

File: src/components/questions/addAnswer.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    let value = event.target.value;
+    this.setState({[feild] : value})
+  }
+
+  checkEmailValidity(email){
+    let regex = RegExp(/^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/);
```
Comment (dprevoznik) [inline src/components/questions/addAnswer.js:30]: You will have to explain this to me one day.

Comment (dprevoznik) [reply src/components/questions/addAnswer.js:30]: I am also unclear on how the test method works.

File: src/components/questions/addAnswer.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    return regex.test(email)
+  }
+
+  submitAnswer(event){
+    // event.preventDefault()
+    if (this.state.answer.length === 0) {
```
Comment (dprevoznik) [inline src/components/questions/addAnswer.js:36]: Maybe think about pulling out this out into a separate function to modularize the code in this file a little more. Great concept though.

File: src/components/questions/addAnswer.js (hunk 3 above)
Comment (iwantmyhatback) [reply src/components/questions/addAnswer.js:30]: mayanne is the Regex expert! 😂

File: src/components/questions/addAnswer.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+          name:this.state.nickname,
+          email:this.state.email
+        }
+      })
+      .then(data=>{
+        console.log(data);
+        console.log(this.state.qID);
+        console.log("*** Successfully Posted Answer to Question ***")
+        this.props.getProductQuestions(this.props.paramsId);
+        this.props.exitClick()
```
Comment (dprevoznik) [inline src/components/questions/addAnswer.js:59]: Good idea for exiting the modal upon successfully posting the result.

File: src/components/questions/addAnswer.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                rows="15"
+                cols="70"
+                maxLength="1000"
+                onChange={this.onChange}
+                placeholder="Your Answer"
+      ></textarea>
+      <br/>
+      {/* DISPLAY WARNING IF ANSWER IS GETTING TOO LONG*/}
+      {this.state.answer.length > 998 ?
+      <React.Fragment><div>Your Answer is Too Long! Please Shorten</div><br/></React.Fragment> :
```
Comment (dprevoznik) [inline src/components/questions/addAnswer.js:86]: Is max length 1000? Maybe rephrase this to say it is close to being too large.

File: src/components/questions/addAnswer.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+             onChange={this.onChange}
+             placeholder="Jack543">
+      </input>
+      {/* DISPLAY WARNING IF NICKNAME IS GETTING TOO LONG*/}
+      {this.state.nickname.length > 58 ?
+      <React.Fragment><div>Your Nickname is Too Long! Please Shorten</div><br/></React.Fragment> :
+      <React.Fragment></React.Fragment>}
+      {/* DISPLAY WARNING IF NICKNAME FEILD IS EMPTY*/}
+      {this.state.requiredNickname ?
+      <React.Fragment><b>Nickname is a Required Feild!!!</b><br/></React.Fragment> :
```
Comment (dprevoznik) [inline src/components/questions/addAnswer.js:109]: Spelling error on field.

File: src/components/questions/addAnswer.js [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+             onChange={this.onChange}
+             placeholder="Jack@email.com">
+      </input>
+      {/* DISPLAY WARNING IF EMAIL IS GETTING TOO LONG*/}
+      {this.state.email.length > 58 ?
+      <React.Fragment><div>Your Email is Too Long! Please Shorten</div><br/></React.Fragment> :
+      <React.Fragment></React.Fragment>}
+      {/* DISPLAY WARNING IF EMAIL FEILD IS EMPTY*/}
+      {this.state.requiredEmail ?
+      <React.Fragment><b>Email is a Required Feild!!!</b><br/></React.Fragment> :
```
Comment (dprevoznik) [inline src/components/questions/addAnswer.js:130]: Spelling error on field.

File: src/components/questions/question.js [hunk 9]
```diff
@@ -1,26 +1,50 @@
...
+  addAnswerClick(event){
+    this.setState({showModal:!this.state.showModal})
   }
 
   render(){
     return (<div className="container">
       <div className="container">
         <div className="title is-inline-block">{`Q: ${this.props.question_body}`}</div>
-        <div className="is-pulled-right is-inline-block">Helpful?({this.props.question_helpfulness}){" | "}Add Answer</div>
+        <div className="is-pulled-right is-inline-block">
```
Comment (dprevoznik) [inline src/components/questions/question.js:29]: I have used is-pulled-right before for certain things. Possibly look into different solution as I noticed some unexpected styling behavior when I used it.

Comment (dprevoznik) [reply src/components/questions/question.js:29]: An alternative may be doing the right alignment in a separate css file and importing it.

File: src/components/questions/question.js [hunk 10]
```diff
@@ -1,26 +1,50 @@
...
     super(props);
-    this.state = {};
+    this.state = {
+      showModal: false
+    };
+    this.helpfulClick = this.helpfulClick.bind(this);
+    this.addAnswerClick = this.addAnswerClick.bind(this);
+  }
+
+  helpfulClick(event){
```
Comment (dprevoznik) [inline src/components/questions/question.js:17]: Interested in how you will track helpfulness click. Do you need to track individual user to make sure they don't report helpfulness twice.

File: src/components/questions/question.js [hunk 11]
```diff
@@ -1,26 +1,50 @@
...
       </div>
-      {/* <div>
-      {this.props.asker_name}{" | "}{this.props.question_date}{" | "}{this.props.reported}
-      </div> */}
       <AnswerList answers={this.props.answers}/>
       <br/>
       <br/>
+      {this.state.showModal ?
+      <Modal title="Submit Your Answer"
+             children={
```
Comment (dprevoznik) [inline src/components/questions/question.js:40]: very interesting that this works. You can also put the implemented code between the opening and closing tags of the modal if you wanted.

File: src/components/questions/questionsList.js [hunk 12]
```diff
@@ -40,19 +39,22 @@ constructor(props){
   render(){
     return (
     <React.Fragment>
-      <h1 className="title">QUESTONS & ANSWERS</h1>
+      <h1 className="title">QUESTIONS & ANSWERS</h1>
       <div className="container questionsExpand">
         {
         (this.props.store.questions[0] ?
           this.determineDisplayed(this.props.store.questions).map( (question, index) => {
```
Comment (dprevoznik) [inline src/components/questions/questionsList.js:46]: I did a similar conditional rendering of data in my list component for rendering my related product cards. Helped simplify the number of steps needed to get the data on the DOM.

**Review by dprevoznik — APPROVED** [review]
This looks great. Excited to checkout what the form looks like and to post some questions!

File: src/components/questions/addAnswer.js (hunk 6 above)
Comment (iwantmyhatback) [reply src/components/questions/addAnswer.js:86]: Max length is 1000 characters but it wont allow input of more than 1000 with the "maxLength" property set so i just set the message to display on 999 as a work around

File: src/components/questions/question.js (hunk 10 above)
Comment (iwantmyhatback) [reply src/components/questions/question.js:17]: yes sir, i believe mayanne has handled this already so i may talk to her about her implementation

File: src/components/questions/question.js (hunk 11 above)
Comment (iwantmyhatback) [reply src/components/questions/question.js:40]: i will prob make that change in the future. i forgot about that.
